### PR TITLE
[serde generate] LCS runtime

### PR DESCRIPTION
**Pull request by matbd**
## Summary

Make available a python LCS runtime as part of Serde generate.

## Test Plan

Included unit tests.

### CODE REVIEW

File: serde-generate/Cargo.toml [hunk 1]
```diff
@@ -28,7 +28,7 @@ serde-reflection = { path = "../serde-reflection", version = "0.3.0" }
 tempfile = "3.1"
 bincode = "1.2"
 hex = "0.4.2"
-
+libra-canonical-serialization = { git = "https://github.com/libra/libra.git", branch = "master" }
```
Comment (thefallentree) [inline serde-generate/Cargo.toml:31]: don't we need testnet branch? otherwise one breaking change (LCS related)  will cause trouble for us.

**Comment by thefallentree** [comment]
my suggestion is that we can make serde-generate 
1.  generate an bunch of fixtures files (random LCS struct with LCS bytes)  
2.  install the generated python lcs runtime
3.  verify the python lcs runtime can deserialize and serialize back to exact bytes

it seems to be an simpler and more generic solution for any runtime we want to support

File: serde-generate/Cargo.toml (hunk 1 above)
Comment (matbd) [reply serde-generate/Cargo.toml:31]: Sure, I can change that.

File: serde-generate/tests/lcs_runtime.rs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+#[test]
+fn test_python_lcs_runtime_on_all_supported_types() {
+    let registry = test_utils::get_registry().unwrap();
+    let dir = tempdir().unwrap();
+    let source_path = dir.path().join("test.py");
+    let mut source = File::create(&source_path).unwrap();
+    python3::output(&mut source, &registry).unwrap();
+
+    let values = test_utils::get_sample_values();
```
Comment (matbd) [inline serde-generate/tests/lcs_runtime.rs:94]: @thefallentree What you are suggesting is happening here and below. The values are not random at the moment but they cover every LCS construction.

Comment (matbd) [reply serde-generate/tests/lcs_runtime.rs:94]: We test the installer separately (verifying that moving files around does not break python import).

File: serde-generate/Cargo.toml (hunk 1 above)
Comment (matbd) [reply serde-generate/Cargo.toml]: done

File: serde-generate/tests/lcs_runtime.rs (hunk 2 above)
Comment (matbd) [reply serde-generate/tests/lcs_runtime.rs:94]: This being said, we could do better on testing. In particular, it's missing negative examples.

**Pull request merged**